tests/git.analysis.spec.ts

```diff
@@ -191,6 +192,39 @@ describe('Functional test of analysis', () => {
   });
 });
 
+describe('Custom request options', () => {
+  beforeAll(() => {
+    jest.mock('axios');
+    axios.request = jest.fn().mockRejectedValue({});
+  });
+
+  it(
+    'passes custom options correctly',
+    async () => {
+      try {
+        await analyzeGit(
+          {
+            baseURL,
+            sessionToken,
+            includeLint: false,
+            severity: 1,
+            gitUri: 'git@github.com:DeepCodeAI/cli.git',
```

Comment:
This makes the test reliant on the network (which means we're ultimately testing the network too) - given what we're trying to achieve here, I guess there's no other way and we can't mock this?

Reply:
@solal-snyk This is the first test, that is mocking network here 😂 We have a lot of network dependent tests there. This one is not using network because of https://github.com/snyk/code-client/pull/64/files/e46f2e315e62fa0da97a7655c2b2515993caa028#diff-470745133bb7400e74c85a9f84a74d1494a4ec040d6d0908e617f8d217132209R198

axios is mocked to reject request immediately, so network doesn't play any role here.

Reply:
Awesome thanks for explaining just learned something very useful regarding our stack 😄